tb_plugin/torch_tb_profiler/plugin.py

```diff
@@ -63,8 +72,10 @@ def __init__(self, context):
         def clean():
             logger.debug("starting cleanup...")
             self._cache.__exit__(*sys.exc_info())
-            for temp_file in self._gpu_metrics_file_dict.values():
-                logger.info("remove temporary file %s with gpu metrics" % temp_file)
+            logger.info("remove temporary cache directory %s" % consts.TEMP_DIR)
```

Comment:
TEMP_DIR should be lazily created by a getter; otherwise it will be created at import (which is not ideal) and the cleanup may or may not be registered. If there is a getter then you can do the atexit register during the lazy init which makes it clear that the temp dir cannot leak.

Reply:
Thanks for your comment. I have submitted another pr for the tempdir deletion logic: https://github.com/pytorch/kineto/pull/470